### PR TITLE
feat: export body of node definition into function

### PR DESCRIPTION
**Pull request by Core5563**

### CODE REVIEW

**Comment by EagleoutIce** [comment]
* [ ] Unify new style with `React.FC` to signal functional components
* [ ] Add edge types (with arrow-heads)

File: source/components/model/edges/edgeDefinition.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    case 'calls': return 'callsEdge'
+    case 'returns': return 'returnsEdge'
+    case 'defines-on-call': return 'definesOnCallEdge'
+    case 'defined-by-on-call': return 'definedByOnCallEdge'
+    case 'argument': return 'argumentEdge'
+    case 'side-effect-on-call': return 'sideEffectOnCallEdge'
+    case 'relates': return 'relatesEdge'
+  }
+  return ''
+}
```
Comment (EagleoutIce) [inline source/components/model/edges/edgeDefinition.tsx:19]: Please use a record like this (you should be able to just apply the suggestion). You can use a map as well if you want:

```suggestion
const edgeTagMap = {
  'reads':               'readsEdge',
  'defined-by':          'definedByEdge',
  'same-read-read':      'sameReadReadEdge',
  'same-def-def':        'sameDefDefEdge',
  'calls':               'callsEdge',
  'returns':             'returnsEdge',
  'defines-on-call':     'definesOnCallEdge',
  'defined-by-on-call':  'definedByOnCallEdge',
  'argument':            'argumentEdge',
  'side-effect-on-call': 'sideEffectOnCallEdge',
  'relates':             'relatesEdge'
};

function edgeTagMapper(edgeTag: string): string {
  return edgeTagMap[edgeTag] ?? '';
}
```

File: source/components/model/edges/edgeDefinition.tsx [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  const [edgePath, labelX, labelY] = getStraightPath({
+    sourceX: props.sourceX ,
+    sourceY: props.sourceY,
+    targetX: props.targetX,
+    targetY: props.targetY,
+  })
+  return (
+    <>
+    <BaseEdge id={props.id} path={edgePath} style={props.edgeStyle} />
+    <EdgeLabelRenderer>
```
Comment (EagleoutIce) [inline source/components/model/edges/edgeDefinition.tsx:42]: We could abstract this as a component as well, but that might be fine later (so we should make an issue for this).

File: source/components/model/edges/edgeDefinition.tsx [hunk 3]
```diff
@@ -0,0 +1,266 @@
...
+
+export function SameReadReadEdge({ id, sourceX, sourceY, targetX, targetY} : {
+  id: string,
+  sourceX: number,
+  sourceY: number,
+  targetX: number,
+  targetY: number
+  }) {
+  return <BodyEdgeCompontent 
+    id = {id}
```
Comment (EagleoutIce) [inline source/components/model/edges/edgeDefinition.tsx:105]: there is a lot of repetition here which we may want to "abstract away"

File: source/components/model/nodes/nodeDefinition.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+interface NodeComponentProps {
+  readonly data: NodeProps['data']
+}
+
+export function BodyNodeComponent(props: React.PropsWithoutRef<BodyNodeComponentProps>){
+  return (
+    <HandleNodeComponent>
+      <div className = {props.className}>
+        <span className='hover-over-text'>
```
Comment (EagleoutIce) [inline source/components/model/nodes/nodeDefinition.tsx:18]: fine for now, and a nice intermediate solution :) 

We could make a "hover-over-component" for this if you want. What is your opinion?

**Review by EagleoutIce — CHANGES_REQUESTED** [review]

File: source/components/model/nodes/nodeDefinition.tsx (hunk 4 above)
Comment (Core5563) [reply source/components/model/nodes/nodeDefinition.tsx:18]: seems like a good idea

File: source/components/model/edges/edgeDefinition.tsx (hunk 2 above)
Comment (Core5563) [reply source/components/model/edges/edgeDefinition.tsx:43]: Ive created an Issue for this: #13

File: source/components/model/edges/edgeDefinition.tsx (hunk 3 above)
Comment (Core5563) [reply source/components/model/edges/edgeDefinition.tsx]: I've created an issue for this: #13

**Pull request merged**